Lib/shutil.py

```diff
@@ -1104,6 +1104,10 @@ def make_archive(base_name, format, root_dir=None, base_dir=None, verbose=0,
     supports_root_dir = getattr(func, 'supports_root_dir', False)
     save_cwd = None
     if root_dir is not None:
+        stmd = os.stat(root_dir).st_mode
+        if not stat.S_ISDIR(stmd):
+            raise NotADirectoryError(root_dir)
```

Comment:
It raises an exception without proper `errno` and `filename` attributes and error message.

You need to provide all of them explicitly: `NotADirectoryError(errno.ENOTDIR, 'Not a directory', root_dir)`.

Or call some OS function which raises an appropriate exception implicitly. For example: `with os.scandir(root_dir): pass`. I am wandering whether `os.stat(os.path.join(root_dir, ''))` has the desired behavior on all platforms.

Add tests for `errno` and `filename` attributes of the exception. The message may be localized, so it is not possible to test it without diving too deep in implementation details.

Reply:
I added the appropriate attrs to the exception. Personally I prefer the explicit approach with `os.stat` considering that it indicates intent better. The code does less things and performance should be slightly better. `make_archive` also won't be affected if there's a bug in the more complex `os` function.

However, please do point out if it would be better to change anything.